src/extensions/default/AutoUpdate/node/AutoUpdateDomain.js

```diff
@@ -167,6 +167,82 @@
 
+	/**
+     * Parse the Installer log and search for a error strings
+	 * one it finds the line which has any of error String
+     * it return that line and exit
+     */
+    function parseInstallerLog(filepath, searchstring, encoding, callback) {
+        var line = "";
+        var    searchFn = function (str) {
+            var arr = str.split('\n'),
+                lineNum,
+                pos;
+            for (lineNum = arr.length - 1; lineNum >= 0; lineNum--) {
+                var searchStrNum;
+                for (searchStrNum = 0; searchStrNum < searchstring.length; searchStrNum++) {
+                    pos = arr[lineNum].search(searchstring[searchStrNum]);
+                    if (pos !== -1) {
+                        line = arr[lineNum];
+                        break;
+                    }
+                }
+                if (pos !== -1) {
+                    break;
+                }
+            }
+            callback(line);
+        };
+
+        fs.readFile(filepath, {"encoding": encoding})
+            .then(function (str) {
+                return searchFn(str);
+            }).catch(function () {
+                callback("");
+            });
+    }
+
+    function checkInstallerStatus(searchParams) {
+        var installErrorStr = searchParams.installErrorStr,
+            bracketErrorStr = searchParams.bracketErrorStr,
+            updateDirectory = searchParams.updateDir,
+            encoding =        searchParams.encoding || "utf8",
+            statusObj = {},
+            logFileAvailable = false;
+
+        var notifyBracket = function (errorline) {
+            statusObj.installError = errorline;
+            postMessageToBrackets(MessageIds.NOTIFY_INSTALLATION_STATUS, statusObj);
+        };
+
+        var parseLog = function (files) {
+            files.forEach(function (file) {
+                var fileExt = path.extname(path.basename(file));
+                if (fileExt === ".logs") {
+                    var fileName = path.basename(file),
+                        fileFullPath = updateDirectory + '/' + file;
+                    if (fileName.search("installStatus.logs") !== -1) {
+                        logFileAvailable = true;
+                        parseInstallerLog(fileFullPath, bracketErrorStr, "utf8", notifyBracket);
+                    } else if (fileName.search("update.logs") !== -1) {
+                        logFileAvailable = true;
+                        parseInstallerLog(fileFullPath, installErrorStr, encoding, notifyBracket);
+                    }
+                }
+            });
+            if (!logFileAvailable) {
+                postMessageToBrackets(MessageIds.NOTIFY_INSTALLATION_STATUS, statusObj);
+            }
+        };
+
+        fs.readdir(updateDirectory)
+            .then(function (files) {
+                return parseLog(files);
+            }).catch(function () {
+                postMessageToBrackets(MessageIds.NOTIFY_INSTALLATION_STATUS, statusObj);
```

Comment:
How are the API failures logged? They seem to be logged as empty strings.

Reply:
I changed now , If there is any API failures we will will be sending an error code "BA_UN". Just to convey that  Install has failed with some unknown reason.